dropwizard-demo/src/main/java/com/dep/resource/Controller.java

```diff
@@ -26,52 +28,49 @@
-        return Response.ok(twitterImplement.fetchLatestTweets()).build();
+    @RequestMapping("/getTweets")
+    public List<TweetResponse> fetchTweets() {
+        return twitterImplement.fetchLatestTweets();
     }
 
-
-    @RequestMapping("/filteredTweets/{searchKey}")
+    @RequestMapping("/filteredTweets")
     @Consumes(MediaType.APPLICATION_JSON)
-    public Response filteredTweets(@PathVariable String searchKey) throws TwitterException {
-        List<TweetResponse> response = twitterImplement.getFilteredTweets(searchKey);
-        return Response.ok(response).build();
+    public List<TweetResponse> filteredTweets(@QueryParam("searchKey") String searchKey) {
+        return twitterImplement.getFilteredTweets(searchKey);
     }
 
-    @RequestMapping("/getPage/start/{start}/size/{size}")
+    @RequestMapping("/getPage")
     @Consumes(MediaType.APPLICATION_JSON)
-    public Response getpage(@PathVariable int start, @PathVariable int size) throws TwitterException {
-        List<TweetResponse> response = twitterImplement.getpage(start, size);
-        return Response.ok(response).build();
+    public List<TweetResponse> getPage(@QueryParam("start") int start, @QueryParam("size") int size) throws TwitterException {
+        return twitterImplement.getpage(start, size);
     }
 
-
     @RequestMapping("/healthCheck")
     public String healthCheck() {
         return "Ping Received at " + new Date();
     }
 
-
-    @RequestMapping(method = RequestMethod.POST, value = "/tweetAgain")
-    public SendResponse sendTweet(@RequestBody Request request) throws TwitterException {
-        logger.info("got into post");
-        String post = request.getMessage();
-        if (StringUtil.isEmpty(post)) {
-            logger.error("error happened");
-            return new SendResponse("Please enter valid tweet", 400);
-        } else {
-            Status status = twitterImplement.sendTweets(post);
-            if (status.getText().equals(post)) {
-                logger.info("successfully posted");
-                return new SendResponse("Tweet posted Successfully", 200);
+    @RequestMapping("/tweetAgain")
+    public SendResponse sendTweet(@RequestBody Request request) {
+        try {
+            logger.info("got into post");
+            String post = request.getMessage();
+            if (StringUtil.isEmpty(post)) {
+                logger.error("error happened");
+                return new SendResponse("Please enter valid tweet", 400);
             } else {
-                logger.error("internal error occurred");
-                return new SendResponse("internal server error", 500);
+                Status status = twitterImplement.sendTweets(post);
+                if (status.getText().equals(post)) {
+                    logger.info("successfully posted");
+                    return new SendResponse("Tweet posted Successfully", 200);
+                } else {
+                    logger.error("internal error occurred");
```

Comment:
how can this be a `internal server error`, program will come to this place only if tweet is not successful from `twitterImplement.sendTweets`, in that case its a problem of twitter library. Also `internal server error` message should be something understandable , its technical term, please update it.

Reply:
Changed the message to "Request is incomplete"